### PR TITLE
Support preferred contact mail on external personnel

### PR DESCRIPTION
**Pull request by HansDahle**
- [x] New feature
- [ ] Bug fix
- [ ] High impact

**Description of work:**
Updated to latest integration lib where preferred contact mail is included. 

Added support for updating preferred contact mail in people service.
Exposed this functionality in simple PUT endpoint which allowes batch update of all personnel. 
Also added support for field on api models and PUT on individual personnel.

Did not add support for including when creating new personnel.

**Testing:**
- [x] Can be tested
- [x] Automatic tests created / updated
- [x] Local tests are passing

Should add automatic tests
Should be able to test using frontend build pointing to this pr backend

**Checklist:**
- [x] Considered automated tests
- [ ] Considered updating specification / documentation
- [ ] Considered work items 
- [ ] Considered security
- [x] Performed developer testing
- [x] Checklist finalized / ready for review

### CODE REVIEW

File: src/backend/api/Fusion.Resources.Api/Controllers/Validation/CustomValidatorExtensions.cs [hunk 1]
```diff
@@ -178,6 +178,54 @@ public static string ToLowerFirstChar(this string input)
...
+        /// </summary>
+        public static IRuleBuilderOptions<T, string?> NotHaveInvalidMailDomain<T>(this IRuleBuilder<T, string?> ruleBuilder)
+        {
+            
+            return ruleBuilder.SetValidator(new EmailDomainValidator())
+                .WithMessage("Mail domain should not be private. Most major private mail domains are rejected (gmail, hotmail, icloud etc.)");
+        }
     }
 
+    public class EmailDomainValidator : PropertyValidator, IPropertyValidator
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Api/Controllers/Validation/CustomValidatorExtensions.cs:193]: Just asking. Do we have a "global" email validator somewhere that you are aware of ?

**Review by xssic — COMMENTED** [review]
Nothing special, just noticed that some tests are failing.

**Review by xssic — COMMENTED** [review]
Github Actions complains about missing exception handling :-)

File: src/backend/api/Fusion.Resources.Api/Controllers/Validation/CustomValidatorExtensions.cs (hunk 1 above)
Comment (HansDahle) [reply src/backend/api/Fusion.Resources.Api/Controllers/Validation/CustomValidatorExtensions.cs:193]: We have a validator that checks that the email is a valid email. Think thats in the Web library. However this is validating for illegal domains in the email...

**Review by xssic — APPROVED** [review]
Looks ok 👍

**Pull request merged**